server/src/main/kotlin/net/horizonsend/ion/server/features/space/encounters/EncounterManager.kt

```diff
@@ -0,0 +1,65 @@
+package net.horizonsend.ion.server.features.space.encounters
+
+import net.horizonsend.ion.server.features.space.generation.SpaceGenerationManager
+import net.horizonsend.ion.server.miscellaneous.NamespacedKeys
+import net.minecraft.nbt.CompoundTag
+import net.minecraft.nbt.NbtIo
+import net.starlegacy.util.toBlockPos
+import org.bukkit.craftbukkit.v1_19_R2.CraftWorld
+import org.bukkit.event.EventHandler
+import org.bukkit.event.Listener
+import org.bukkit.event.player.PlayerInteractEvent
+import org.bukkit.persistence.PersistentDataType
+import java.io.ByteArrayInputStream
+
+object EncounterManager : Listener {
+	@EventHandler
+	fun onPlayerInteract(event: PlayerInteractEvent) {
+		val clickedBlock = event.clickedBlock ?: return
+
+		val serverLevel = (event.player.world as CraftWorld).handle
+		// Quick check if the world of the event would contain a wreck
+		if (!SpaceGenerationManager.worldGenerators.containsKey(serverLevel)) return
+
+		val pdc = event.clickedBlock!!.location.chunk.persistentDataContainer.get(
+			NamespacedKeys.WRECK_ENCOUNTER_DATA,
+			PersistentDataType.BYTE_ARRAY
+		) ?: return
+
+		// Check if the chunk of the event contains an encounter, if not, return
+
+		val wreckData = try { // get existing asteroid data
+			NbtIo.readCompressed(
```

Comment:
dont compres

Reply:
Need to look into this. The compression does cause quite a bit of lag, but the normal read function uses DataInput instead of Input streams and idk how to use those yet.